app/src/main/java/com/example/activity/FirstSevice.java

```diff
@@ -40,10 +66,16 @@ public int onStartCommand(Intent intent, int flags, int startId)
         return super.onStartCommand(intent, flags, startId);
     }
 
-    void todoFunc()
+    private void todoFunc()
     {
         SystemClock.sleep(300);
```

Comment:
Actually here you block UI thread when calling SystemClock.sleep() for 300 ms as Service run in main thread by default. You should process any long-running operations in worker thread.

Reply:
Done